lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -183,7 +185,16 @@ public List<BufferData> getBuffer()
     {
         checkState(closed);
         DataStreams dataStreams = getDataStreams();
-        return ImmutableList.of(new BufferData(dataStreams.data(), dataStreams.dictionaryPageSize(), dataStreams.bloomFilter(), getColumnMetaData()));
+        ColumnMetaData columnMetaData = getColumnMetaData();
+
+        EncodingStats stats = convertEncodingStats(columnMetaData.getEncoding_stats());
+        boolean isOnlyDictionaryEncodingPages = stats.hasDictionaryPages() && !stats.hasNonDictionaryEncodedPages();
+
+        return ImmutableList.of(new BufferData(
+                dataStreams.data(),
+                dataStreams.dictionaryPageSize(),
+                isOnlyDictionaryEncodingPages ? Optional.empty() : dataStreams.bloomFilter(),
```

Comment:
Why we can't have both always? Seems like we should be able to perform bloom filtering even in the presence of dictionaries?

Reply:
For fully dictionary encoded columns, the reader is already able to perform row-group pruning based on dictionary entries. Having a bloom filter doesn't give us anything extra.

Reply:
> For fully dictionary encoded columns, the reader is already able to perform row-group pruning based on dictionary entries

I thought that bloom filters could be more efficient at filtering for tactical (e.g. single value search) queries.

Reply:
It's lookup in a set vs lookup in a bloom filter. The CPU difference will be barely noticeable compared to everything else that goes on in parquet reader. Reading bloom filter takes extra reads from file and potentially has false positives.